packages/core/README.md

```diff
@@ -4,45 +4,51 @@
 
 ## Usage
 
-#### Inputs/Outputs
-
-You can use this library to get inputs or set outputs:
+### Import the package
 
 ```js
+// javascript
 const core = require('@actions/core');
 
-const myInput = core.getInput('inputName', { required: true });
+// typescript
+import * as core from '@actions/core';
+```
 
-// Do stuff
+#### Inputs/Outputs
+
+Action inputs can be read with `getInput`.  Outputs be set with `setOutput` which makes them available to be mapped into inputs of other actions to ensure they are decoupled.
+
+```js
+const myInput = core.getInput('inputName', { required: true });
 
 core.setOutput('outputKey', 'outputVal');
 ```
 
 #### Exporting variables
 
-You can also export variables for future steps. Variables get set in the environment.
+Since steps run out of proc, you can use `exportVariable` to add it to this step and future steps environment blocks.
```

Comment:
Might be worthwhile to define out of proc (I believe its not that common of a terminology)
"Steps do not share environmental variables because each step executes in a different process (out-of-proc). You can use `exportVariable` to add a variable to the current step and all remaining steps in the job."